src/StructuredLogger/Strings/Strings.cs

```diff
@@ -344,14 +344,21 @@ private static string GetPropertyReassignmentText()
             return "^" + text + "$";
         }
 
-        public static Regex CreateRegex(string text, int replacePlaceholders = 0, RegexOptions options = RegexOptions.Compiled)
+        public static Regex CreateRegex(string text, int replacePlaceholders = 0, RegexOptions options = RegexOptions.Compiled | RegexOptions.Singleline, bool capture = false)
         {
-            text = Regex.Escape(text);
+            text = "^" + Regex.Escape(text) + "$";

```

Comment:
I think we should check if these two symbols are already present. Also not sure if it's always safe to add these? I hope so?

Reply:
Added check.  Though I think the Escape would invalidate it.